### ISSUE TITLE
Are we ought to use json-refs?

### ISSUE THREAD

**Issue by PavelPolyakov** [mohsen1/multi-file-swagger-example#4]
Hi @mohsen1 ,

We've started to use swagger and want to separate our definitions. Have some question regarding your article.

I have cloned your repository and put index.yaml directly to the swagger ui (everything is served, using gulp connect). However it results to the error:
![image](https://cloud.githubusercontent.com/assets/839290/12356954/a3864a3e-bba5-11e5-8c41-138cfad49aaf.png)

Does that mean that we should only use the refs with ```json-refs``` and processing it to the one file?
Or it's only an option.

Regards,

**Comment by mohsen1** [comment]
Yes, SwaggerUI does not resolve all the `$refs`. It's much easier to use `json-refs` to resolve all the refs before passing it to SwaggerUI.

**Comment by PavelPolyakov** [comment]
well it resolves, but not completely, I would say :) 

Thanks for the article and for the answer!

**Comment by chasetec** [comment]
You really should be clear that the split multi-file input is not valid Swagger and that you are making a JSON pointer pre-processing tool that can output valid Swagger.

Swagger allows very limited use of $ref. It can only be used in pathItem, schema, responseValue, parametersList. The example from the blog post such as the `info` example are not valid Swagger until after pre-processing.

**Comment by mohsen1** [comment]
@chasetec you are correct. Feel free to fork and make a PR to add this clarification to this repo
